addons/calendar/tests/test_event_recurrence.py

```diff
@@ -352,6 +352,26 @@ def test_backward_pass_dst(self):
             (datetime(2023, 4, 27, 7, 00), datetime(2023, 4, 27, 8, 00)),
         ])
 
+    def test_all_day_date(self):
+        recurrence = self.env['calendar.event'].with_context(
+            default_start_date=date(2019, 10, 22),
+            default_stop_date=date(2019, 10, 22),
+        ).create({
+            'name': 'Recurrent Event',
+            'start': datetime(2019, 10, 22, 8, 0),
+            'stop': datetime(2019, 10, 22, 18, 0),
+            'recurrency': True,
+            'rrule_type': 'weekly',
+            'tue': True,
+            'interval': 1,
+            'count': 2,
+            'event_tz': 'UTC',
+            'allday': True,
+        }).recurrence_id
+        events = recurrence.calendar_event_ids
+        self.assertEqual(events[1].start_date, date(2019, 10, 29))
+        self.assertEqual(events[1].stop_date, date(2019, 10, 29))
```

Comment:
you can provide a small explanation as the last argument of assertEqual.
It can be used to explain the purpose of the check.
I would add:
self.assertEqual(events[0].start_date, date(2019, 10, 22), "The first event is at the start_date parameter")
self.assertEqual(events[1].start_date, date(2019, 10, 29), "the second event is one week after")
it helps bugfix and other devs to understand what we are trying to do when they break the test by accident.